funding/commitment_type_negotiation.go

```diff
@@ -8,12 +8,6 @@ import (
 )
 
 var (
-	// errUnsupportedExplicitNegotiation is an error returned when explicit
-	// channel commitment negotiation is attempted but either peer of the
-	// channel does not support it.
-	errUnsupportedExplicitNegotiation = errors.New("explicit channel " +
```

Comment:
Not sure about the removal of this error. In this case the sender will get less explicit error?

Reply:
The error only exists due to a previous bandaid-type fix: https://github.com/lightningnetwork/lnd/pull/6027.

After the fix in this PR, there seems to be little need for the error message.  The only way to get non-default channel types is through explicit negotiation.  If the peer doesn't support explicit negotiation it also won't support the desired channel type, so it seems just as accurate to return `errUnsupportedChannelType`.

The one case where it would potentially make sense to return `errUnsupportedExplicitNegotiation` is when the default channel type matches the desired channel type *and* the peer doesn't support explicit negotiation.  But this PR makes handles that case by using implicit negotiation instead of failing like we did previously.